pkg/internal/approver/validation/serviceaccount.go

```
@@ -0,0 +1,163 @@
+	if reflect.TypeOf(sa).AssignableTo(typeDesc) {
+		return sa, nil
+	}
+	if reflect.TypeOf("").AssignableTo(typeDesc) {
+		return sa, nil
+	}
+	return nil, fmt.Errorf("type conversion error from 'serviceaccount' to '%v'", typeDesc)
+}
+
+// ConvertToType implements ref.Val.ConvertToType.
+func (sa ServiceAccount) ConvertToType(typeVal ref.Type) ref.Val {
+	switch typeVal {
+	case SAType:
+		return sa
+	case types.TypeType:
+		return SAType
+	}
+	return types.NewErr("type conversion error from '%s' to '%s'", SAType, typeVal)
+}
+
+// Equal implements ref.Val.Equal.
+func (sa ServiceAccount) Equal(other ref.Val) ref.Val {
+	otherDur, ok := other.(ServiceAccount)
+	if !ok {
+		return types.MaybeNoSuchOverloadErr(other)
+	}
+	return types.Bool(sa.Name == otherDur.Name && sa.Namespace == otherDur.Namespace)
+}
+
+// Type implements ref.Val.Type.Y
+func (sa ServiceAccount) Type() ref.Type {
+	return SAType
+}
+
+// Value implements ref.Val.Value.
+func (sa ServiceAccount) Value() interface{} {
+	return sa
+}
+
+var saLibraryDecls = map[string][]cel.FunctionOpt{
+	"serviceAccount": {
+		cel.Overload("username_to_serviceaccount", []*cel.Type{cel.StringType}, SAType,
+			cel.UnaryBinding(stringToServiceAccount))},
+	"getName": {
+		cel.MemberOverload("serviceaccount_get_name", []*cel.Type{SAType}, cel.StringType,
+			cel.UnaryBinding(getServiceAccountName))},
+	"getNamespace": {
+		cel.MemberOverload("serviceaccount_get_namespace", []*cel.Type{SAType}, cel.StringType,
+			cel.UnaryBinding(getServiceAccountNamespace))},
+	"isServiceAccount": {
+		cel.Overload("serviceaccount_is_sa", []*cel.Type{cel.StringType}, cel.BoolType,
+			cel.UnaryBinding(isServiceAccount))},
+}
+
+func stringToServiceAccount(arg ref.Val) ref.Val {
+	s, ok := arg.Value().(string)
+	if !ok {
+		return types.MaybeNoSuchOverloadErr(arg)
+	}
+
+	ns, name, err := serviceaccount.SplitUsername(s)
+
+	if err != nil {
+		return ServiceAccount{
```

Shouldn't we return an error here instead? An error here will be extremely hard to debug with the current lenient handling.